### PR TITLE
🧹 Clean Up Tests + More Explicit Example

### PR DESCRIPTION
**Pull request by jayceslesar**
Little work to clean up tests with pytest + combine tests that could be combined.

Also added a few helpful lines to the README for clarity

### CODE REVIEW

**Comment by jvdd** [comment]
Nice parametrization of the tests @jayceslesar!! 
I'll try to merge it later today :)

File: tests/test_tsdownsample.py [hunk 1]
```diff
@@ -9,99 +11,76 @@
...
+    MinMaxLTTBDownsampler(),
+]
 
-def test_minmax_downsampler():
-    """Test MinMax downsampler."""
-    arr = np.array(np.arange(10_000))
-    s_downsampled = MinMaxDownsampler().downsample(arr, n_out=100)
-    assert s_downsampled[0] == 0
-    assert s_downsampled[-1] == len(arr) - 1
+ALL_DOWNSAMPLERS = [EveryNthDownsampler()]
```
Comment (jvdd) [inline tests/test_tsdownsample.py:27]:
```suggestion
ALL_DOWNSAMPLERS = RUST_DOWNSAMPLERS + [EveryNthDownsampler()]
```

Comment (jvdd) [inline tests/test_tsdownsample.py:27]: Perhaps rename to `OTHER_DOWNSAMPLERS`?

Comment (jayceslesar) [reply tests/test_tsdownsample.py:27]: Makes sense

File: tests/test_tsdownsample.py [hunk 2]
```diff
@@ -9,99 +11,81 @@
...
 
 def test_everynth_downsampler():
     """Test EveryNth downsampler."""
-    arr = np.array(np.arange(10_000))
-    s_downsampled = EveryNthDownsampler().downsample(arr, n_out=100)
+    arr = np.arange(10_000)
+    downsampler = EveryNthDownsampler()
+    s_downsampled = downsampler.downsample(arr, n_out=100)
     assert s_downsampled[0] == 0
     assert s_downsampled[-1] == 9_900
```
Comment (jvdd) [inline tests/test_tsdownsample.py:55]: Having a seperate test is imo cleaner than having an if-case in the `test_rust_downsampler` (besides that, EveryNth is also not a rust downsampler)

**Review by jvdd — APPROVED** [review]
Very pleased with the improvements to the tests! Thanks for contributing @jayceslesar :rocket:

**Pull request merged**